core/src/main/java/io/questdb/griffin/engine/functions/window/AvgDoubleWindowFunctionFactory.java

```diff
@@ -611,7 +615,14 @@ public void computeNext(Record record) {
                     firstIdx = newFirstIdx;
                 }
 
-                avg = (frameSize != 0 ? sum / frameSize : Double.NaN);
+                if (frameSize != 0) {
```

Comment:
because avg() is updated in this PR, i consider bugs in `avg()` relevant to this PR too :)

`close()` method is a bit of spaghetti, it is never called and instead memory is closed in `reset()`. Could you review this flow and make sure close is indeed called

Reply:
Sure, thank you !

I inherited `reset()` from previous CachedWindowRecordCursorFactory and do agree it's confusing.
`reset()`
At the moment `close()` is only called for window functions when optimized path fails in SqlCodeGenerator . 
Should I replace `reset()` with `close()` ?

Reply:
I would let you decide, they should be one place and if you pick `reset` it would be good to have comments as to why it is enough to release memory there :)

Reply:
Turns out `reset()` and `close()` have slightly different purpose. 
`reset()` is used to release the native memory but keep function usable, so it keeps e.g. argument and partition keys. 
Once `close()` is called (on factory close), function can't be used anymore .